libraries/core_libs/network/src/tarcap/packets_handlers/pbft_sync_packet_handler.cpp

```diff
@@ -57,6 +60,14 @@ void PbftSyncPacketHandler::process(const PacketData &packet_data, const std::sh
     throw MaliciousPeerException("Unable to parse SyncBlock: " + std::string(e.what()));
   }
 
+  std::vector<std::shared_ptr<Vote>> cert_votes;
+  if (pbft_chain_synced) {
```

Comment:
isnt `const bool pbft_chain_synced = packet_data.rlp_[0].toInt<bool>();` redundant ? You can say if the packet is the last one based on item count now ?

Reply:
I guess so although it seems cleaner this way because we are not using some magic number for this. It does not add much to the packet size. But yes I could remove it if you think that is better

Reply:
the problem I see here is that we are setting 2 independent data fields and it could break processing. Someone could create a bug when he set this flag to true but does not provide the the current block cert votes and in such case we would try to process non-existent data field. I would definitely create `standard_size & last_block_size` as class constants and create flag for this if like this: `is_last_block = itemsCount == last_block_size`

Reply:
Ok, I implemented it this way now

Reply:
actually still working on this, my initial change was not good

Reply:
Fixed